src/store/token.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+export async function getToken(token) {
```

Comment:
You could model this after `Array.from()` and make this a static `from()` method on the `Token` class (e.g. `Token.from()`);

Reply:
This would mean you'd need to export the `Token` class, though, which you might not want as you can't make the constructor private and enforce the use of this factory function.

Reply:
Alternatively, you could make getting a token the problem of the caller of this module (i.e. they should already have a token in hand so if they've got a function instead of a value it's their responsibility to invoke the function). This would allow this file to be solely about the `Token` class.

Reply:
The person with the token, won't know what type of token they will have and it would be a mess if everyone had to test for it.

That is why I kept it outside the class so I could export it and control the interface.